lexibank_wangbcd.py

```diff
@@ -1,81 +1,68 @@
-            'Supplementary_Material_I/data/')
-        self.raw.download_and_unpack(
+    def cmd_download(self, args):
+        self.raw_dir.download_and_unpack(
             "https://zenodo.org/record/16760/files/"
             "Network-perspectives-on-Chinese-dialect-history-1.zip",
-            d.joinpath('chinese.tsv'),
-            d.joinpath('old_chinese.csv'),
-            log=self.log)
-        self.raw.write('sources.bib', getEvoBibAsBibtex('Hamed2006', 'List2015d', **kw))
+            self.raw_dir.joinpath("chinese.tsv"),
+            self.raw_dir.joinpath("old_chinese.csv"),
+            log=args.log,
+        )
 
-    def split_forms(self, row, value):
-        """
-        We make sure to always only yield one form per raw lexeme.
-        """
-        return BaseDataset.split_forms(self, row, value)[:1]
+        self.raw_dir.write("sources.bib", getEvoBibAsBibtex("Hamed2006", "List2015d"))
 
-    def cmd_install(self, **kw):
-        wl = lingpy.Wordlist(self.raw.posix('chinese.tsv'))
+    def cmd_makecldf(self, args):
+        wl = lingpy.Wordlist(self.raw_dir.joinpath("chinese.tsv").as_posix())
         maxcogid = 0
 
-        with self.cldf as ds:
-            ds.add_sources()
-            ds.add_languages(id_factory=lambda l: l['Name'])
-            ds.add_concepts(id_factory=lambda c: slug(c.label, lowercase=False))
+        args.writer.add_sources()
+        args.writer.add_languages(id_factory=lambda l: l["Name"])
+        args.writer.add_concepts(id_factory=lambda c: slug(c.label, lowercase=False))
 
-            # store list of proto-form to cognate set
-            p2c = {}
+        # store list of proto-form to cognate set
+        p2c = {}
 
-            for k in wl:
-                for row in ds.add_lexemes(
-                    Language_ID=wl[k, 'doculect'],
-                    Parameter_ID=slug(wl[k, 'concept'], lowercase=False),
-                    Value=wl[k, 'ipa'],
-                    Source='Hamed2006',
-                    Cognacy=wl[k, 'COGID']
+        for k in wl:
+            for row in args.writer.add_lexemes(
+                Language_ID=wl[k, "doculect"],
+                Parameter_ID=slug(wl[k, "concept"], lowercase=False),
+                Value=wl[k, "ipa"],
+                Source="Hamed2006",
+                Cognacy=wl[k, "COGID"],
+            ):
+                args.writer.add_cognate(
+                    lexeme=row, Cognateset_ID=wl[k, "cogid"], Source=["Hamed2006"]
+                )
+            maxcogid = max([maxcogid, int(wl[k, "cogid"])])
+            p2c[wl[k, "proto"]] = wl[k, "cogid"]
+        idx = max([k for k in wl]) + 1
+        for line in lingpy.csv2list(self.raw_dir.joinpath("old_chinese.csv").as_posix()):
+            for val in line[1].split(", "):
+                cogid = p2c.get(val)
+                if not cogid:
+                    maxcogid += 1
+                    cogid = p2c[val] = maxcogid
+                for row in args.writer.add_lexemes(
```

Comment:
`add_lexemes` should be modified by its more explicit counterpart. This is a case where I think we don't need a form spec, so we can just add a form. This makes more sense, I think.

Reply:
Sorry, what do you mean by more explicit counterpart?

Reply:
`add_lexemes` is deprecated, right? you should not use it, but write `add_forms_from_value` but this does not make sense, as the data already comes along as a form, due to the wordlist. So you can use the more explicit command `add_form`.

Reply:
In general, `add_lexemes` should no longer be used.

Reply:
It's an alias for `add_forms_from_value`: https://github.com/lexibank/pylexibank/blob/a6df3d6e1a99606e8ca56896b8fc0deb02040614/src/pylexibank/cldf.py#L212

Reply:
Yeah, that's why I was confused, because it is just an alias. Wasn't aware that this is deprecated, sorry. Will change.

Reply:
And since it does the same as `add_form` here, given that there's no splitting of the forms, I'd just add the forms directly with `add_form`.